### PR TITLE
Refactor keyword and report

### PR DESCRIPTION
**Pull request by minsikzzang**
Refactor keyword and report
* Add examples for keyword and report
* Refactor keyword and report services

### CODE REVIEW

File: examples/report.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
+require 'sogou/search/api/auth'
+require 'sogou/search/api/service/report'
+
+# ENV['SOGOU_API_TOKEN'] = 'xxxx'
```
Comment (chiting) [inline examples/report.rb]: do we really need to keep these comments?

File: lib/sogou/search/api/core/base_service.rb [hunk 2]
```diff
@@ -36,7 +39,10 @@ def client
                 namespaces: { 'xmlns:v1' => 'http://api.sogou.com/sem/common/v1' },
                 env_namespace: :soapenv,
                 namespace_identifier: :v11,
-                soap_header: soap_header
+                soap_header: soap_header,
+                #adapter: :httpclient
```
Comment (chiting) [inline lib/sogou/search/api/core/base_service.rb]: Are these still needed?

File: lib/sogou/search/api/service/report.rb [hunk 3]
```diff
@@ -9,32 +9,32 @@ def initialize
...
-                performance_data: fields,
-                start_date: start_date,
-                end_date: end_date,
-                report_type: type
-              }
-            })
-            puts resp.header
-            resp.body
+          def get_report_id(start_date, end_date, type, fields, options = {}, &block)
+            command = make_command(:get_report_id,
```
Comment (chiting) [inline lib/sogou/search/api/service/report.rb]: if you put the first parameter on the next line you wouldn't have to do such indentation :)

File: lib/sogou/search/api/service/keyword.rb [hunk 4]
```diff
@@ -9,18 +9,30 @@ def initialize
...
+                                     cpc_grp_ids: grp_ids,
+                                     get_temp: valid_keyword_only?(valid_keyword_only)
+                                   },
+                                   options: options)
+            execute_command(command, &block)
+          end
+
+          private
+
+          def valid_keyword_only?(only)
```
Comment (chiting) [inline lib/sogou/search/api/service/keyword.rb]: How come a method with a `?` signature would return `0` or `1`?

Comment (minsikzzang) [reply lib/sogou/search/api/service/keyword.rb]: :) it doesn't seem right

File: lib/sogou/search/api/service/report.rb (hunk 3 above)
Comment (minsikzzang) [reply lib/sogou/search/api/service/report.rb]: hmm, it doesn't look pretty :(

File: examples/report.rb (hunk 1 above)
Comment (minsikzzang) [reply examples/report.rb]: yes, Perhaps, I will add more comment for them

**Pull request merged**